File: System/Spark/Basic/PairFunction/counter.py
from pyspark import SparkConf,SparkContext

import sys
reload(sys)
sys.setdefaultencoding('utf-8')

conf = SparkConf().setMaster("local").setAppName("Pair")
sc = SparkContext(conf = conf)

# word counter
text = sc.textFile("../example.md")
words = text.flatMap(lambda x : x.split(" "))
result = words.map(lambda x : (x,1)).reduceByKey(lambda x,y:x+y).sortByKey()

for i in result.collect():
	print i 

sc.stop()

File: Framework/Spark/Basic/PairFunction/counter.scala
import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._

val conf = new SparkConf().setMaster("local").setAppName("My App")
val sc = new SparkContext(conf)

// Create RDD
val inputRDD = sc.textFile("../example.md")

// Word Counter
val words = inputRDD.flatMap(x => x.split(" "))
val result = words.map(x => (x,1)).reduceByKey((x,y) => x+y)
result.collect().foreach(println)